core/templates/dev/head/domain/exploration/AnswerGroupObjectFactory.js

```diff
@@ -18,9 +18,10 @@
  */
 
 oppia.factory('AnswerGroupObjectFactory', [function() {
-  var AnswerGroup = function(ruleSpecs, outcome, correct) {
-    this.ruleSpecs = ruleSpecs;
-    this.outcome = outcome;
+  var AnswerGroup = function(ruleSpecsBackendList, outcomeBackendDict,
+    correct) {
+    this.ruleSpecs = ruleSpecsBackendList;
```

Comment:
(I kinda like ending with List, Dict, etc. since those sort of indicate that the items are JS lists and dicts and not actual domain objects.)